assets/chat/js/formatters.js

```diff
@@ -118,10 +142,33 @@ class EmoteFormatter {
             const suffixes = Object.keys(GENERIFY_OPTIONS).join('|');
             this.regex = new RegExp(`(^|\\s)(${emoticons})(:(${suffixes}))?(?=$|\\s)`, 'gm');
         }
+
+        if (!this.emotewidths) {
+            this.emotewidths = {};
+            this.emoteheiths = {};
+            const emoteArray = [...chat.emoticons]
```

Comment:
do we need to create a new array? can we just iterate through the `Set` like we do elsewhere (i.e. [here](https://github.com/MemeLabs/chat-gui/blob/master/assets/chat/js/chat.js#L451-L453) or [here](https://github.com/MemeLabs/chat-gui/blob/master/assets/chat/js/chat.js#L296))?

Reply:
I tried both and I tried them just again, when I use those methods and try to console.log(e) something is broken. The site stays black and nothing is outputted to the console.